parser/prism/Translator.cc

```diff
@@ -1674,14 +1676,22 @@ unique_ptr<parser::Node> Translator::translateConst(PrismLhsNode *node) {
     auto location = translateLoc(node->base.location);
     auto name = parser.resolveConstant(node->name);
 
-    if (isInMethodDef) { // Check if this is a dynamic constant assignment (SyntaxError at runtime)
-        // This is a copy of a workaround from `Desugar.cc`, which substitues in a fake assignment,
-        // so the parsing can continue. See other usages of `dynamicConstAssign` for more details.
+    if (!skipDynamicConstantWorkaround) {
```

Comment:
Can we nest this down into the other `if`?

```
if (isInMethodDef && !skipDynamicConstantWorkaround) {
```

This way it's striped from all the copies of this function that don't match that `if constexpr`.

Plus it decreases a level of indenting.